build.gradle.kts

```diff
@@ -150,6 +156,44 @@ tasks.register<Jar>("cleanShadowJar") {
     }
 }
 
+tasks.register<Jar>("shadeSourcesJar") {
+//    archiveClassifier.set("shaded-sources")
+    archiveClassifier.set("sources")
+    dependsOn("sourcesJar")
+
+    val shadingOriginPath: String = shadingOriginPackage.replace('.', '/')
+    val shadingDestinationPath: String = shadingDestinationPackage.replace('.', '/')
+    val tmpShadowedSourcesPath = "${buildDir}/tmp/shadowedSources"
+
+    val sourceCharset: java.nio.charset.Charset = StandardCharsets.UTF_8
+    val shadedSourcesDir: java.nio.file.Path = file(tmpShadowedSourcesPath).toPath()
+
+    doFirst {
+        zipTree(tasks["sourcesJar"].outputs.files.asPath).visit {
+            val relativePath = this.relativePath.toString()
+
+            if (this.file.isDirectory) {
+                return@visit
```

Comment:
Just for my own understanding, what is @visit? I haven't seen this before

Reply:
In Kotlin, if we have nested functions, the `@` helps explicitly state which nested function to return from. 
Found a decent explanation here https://stackoverflow.com/questions/40160489/kotlin-whats-does-return-mean